src/Sparrow/Memory.cs

```diff
@@ -242,5 +246,22 @@ public static void SetInline(byte* dest, byte value, long n)
             Finish:
             ;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Discard(void* baseAddress, long size)
+        {
+            if (PlatformDetails.CanDiscardMemory)
+            {
+                // We explicitely ignore the return codes because even if it fails, from all uses and purposes we dont care. 
```

Comment:
I think that, for debug only, we should error if we fail here.
That is important if we pass bad information, otherwise we might miss this entirely

Reply:
Certain file systems do not support o_direct, but we still want to support then.
For example cifs in docker